realm/src/main/java/io/realm/Realm.java

```diff
@@ -123,25 +122,33 @@
 public final class Realm implements Closeable {
     public static final String DEFAULT_REALM_NAME = "default.realm";
 
-    protected static final ThreadLocal<Map<String, Realm>> realmsCache = new ThreadLocal<Map<String, Realm>>() {
-        @SuppressLint("UseSparseArrays")
+    protected static final ThreadLocal<Map<RealmConfiguration, Realm>> realmsCache =
+            new ThreadLocal<Map<RealmConfiguration, Realm>>() {
         @Override
-        protected Map<String, Realm> initialValue() {
-            return new HashMap<String, Realm>(); // On Android we could use SparseArray<Realm> which is faster,
-                                                  // but incompatible with Java
+        protected Map<RealmConfiguration, Realm> initialValue() {
+            return new HashMap<RealmConfiguration, Realm>();
         }
     };
-    private static final ThreadLocal<Map<String, Integer>> referenceCount = new ThreadLocal<Map<String,Integer>>() {
-        @SuppressLint("UseSparseArrays")
+
+    private static final ThreadLocal<Map<RealmConfiguration, Integer>> referenceCount =
+            new ThreadLocal<Map<RealmConfiguration,Integer>>() {
         @Override
-        protected Map<String, Integer> initialValue() {
-            return new HashMap<String, Integer>();
+        protected Map<RealmConfiguration, Integer> initialValue() {
+            return new HashMap<RealmConfiguration, Integer>();
         }
     };
-    // Maps ids to a boolean set to true if the Realm is open. This is only needed by deleteRealmFile
-    private static final Map<String, AtomicInteger> openRealms = new ConcurrentHashMap<String, AtomicInteger>();
-    private static final int REALM_CHANGED = 14930352; // Hopefully it won't clash with other message IDs.
+
+    // Map between all Realm file paths and all known configurations pointing to that file.
+    private static final Map<String, List<RealmConfiguration>> globalPathConfigurationCache =
```

Comment:
Because we need to detect conflicting configurations for the same path, ie. two different schemas for the same path or two different schema versions, but eg. having one configuration readonly and another as writeable is OK.

So we need to group configurations pr. path to be able to validate them.